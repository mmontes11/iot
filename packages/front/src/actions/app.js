import {
  TOGGLE_HAMBURGER_MENU,
  TOGGLE_MAP_DIALOG,
  TOGGLE_LANGUAGE_SELECTOR,
  SET_SHOW_ERROR,
} from "constants/actionTypes/app";

export const toggleHamburgerMenu = () => dispatch => {
  dispatch({ type: TOGGLE_HAMBURGER_MENU });
};

export const toggleMapDialog = () => dispatch => {
  dispatch({ type: TOGGLE_MAP_DIALOG });
};

export const toggleLanguageSelector = () => dispatch => {
  dispatch({ type: TOGGLE_LANGUAGE_SELECTOR });
};

export const setShowError = showError => dispatch => {
  dispatch({ type: SET_SHOW_ERROR, showError });
};
