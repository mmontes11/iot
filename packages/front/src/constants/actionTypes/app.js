export const TOGGLE_HAMBURGER_MENU = "TOGGLE_HAMBURGER_MENU";
export const TOGGLE_MAP_DIALOG = "TOGGLE_MAP_DIALOG";
export const SET_SHOW_ERROR = "SET_SHOW_ERROR";
