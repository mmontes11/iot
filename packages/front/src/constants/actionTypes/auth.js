export const IS_AUTH = "IS_AUTH";
export const SET_USERNAME = "SET_USERNAME";
export const SET_PASSWORD = "SET_PASSWORD";
export const LOGIN_REQUEST = "LOGIN_REQUEST";
export const LOGIN_REQUEST_SUCCESS = "LOGIN_REQUEST_SUCCESS";
export const LOGIN_REQUEST_ERROR = "LOGIN_REQUEST_ERROR";
