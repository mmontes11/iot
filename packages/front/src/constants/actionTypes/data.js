export const DATA_UPDATED = "DATA_UPDATED";
export const DATA_REQUEST = "DATA_REQUEST";
export const DATA_REQUEST_SUCCESS = "DATA_REQUEST_SUCCESS";
export const DATA_REQUEST_ERROR = "DATA_REQUEST_ERROR";
export const ADD_DATA_ITEM = "ADD_DATA_ITEM";
export const RESET_DATA = "RESET_DATA";
