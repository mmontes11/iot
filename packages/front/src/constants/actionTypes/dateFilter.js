export const DATE_FILTER_SELECTOR_TOGGLE = "DATE_FILTER_SELECTOR_TOGGLE";
export const DATE_FILTER_TIME_PERIOD_SELECT = "DATE_FILTER_TIME_PERIOD_SELECT";
export const DATE_FILTER_TIME_PERIOD_UPDATED = "DATE_FILTER_TIME_PERIOD_UPDATED";
export const DATE_FILTER_TIME_PERIODS_REQUEST = "DATE_FILTER_TIME_PERIODS_REQUEST";
export const DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS = "DATE_FILTER_TIME_PERIODS_REQUEST_SUCCESS";
export const DATE_FILTER_TIME_PERIODS_REQUEST_ERROR = "DATE_FILTER_TIME_PERIODS_REQUEST_ERROR";
export const DATE_FILTER_TIME_PERIODS_UPDATED = "DATE_FILTER_TIME_PERIODS_UPDATED";
export const DATE_FILTER_START_DATE_UPDATED = "DATE_FILTER_START_DATE_UPDATED";
export const DATE_FILTER_END_DATE_UPDATED = "DATE_FILTER_END_DATE_UPDATED";
