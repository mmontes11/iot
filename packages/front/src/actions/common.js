import { RESET } from "constants/actionTypes/common";

export const reset = preserveError => dispatch => {
  dispatch({ type: RESET, preserveError });
};
