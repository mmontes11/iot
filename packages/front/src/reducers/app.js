import { TOGGLE_HAMBURGER_MENU, TOGGLE_MAP_DIALOG, SET_SHOW_ERROR } from "constants/actionTypes/app";
import { RESET } from "constants/actionTypes/common";

export const initialState = {
  isHamburgerMenuExpanded: false,
  isMapDialogOpened: false,
  showError: true,
};

export default (state = initialState, { type, showError, preserveError }) => {
  if (/_REQUEST_ERROR$/.test(type)) {
    return { ...state, showError: true };
  }
  switch (type) {
    case TOGGLE_HAMBURGER_MENU:
      return { ...state, isHamburgerMenuExpanded: !state.isHamburgerMenuExpanded };
    case TOGGLE_MAP_DIALOG:
      return { ...state, isMapDialogOpened: !state.isMapDialogOpened };
    case SET_SHOW_ERROR:
      return { ...state, showError };
    case RESET:
      if (preserveError) {
        return { ...initialState, showError: state.showError };
      }
      return initialState;

    default:
      return state;
  }
};
