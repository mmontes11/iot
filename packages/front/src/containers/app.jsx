import React, { Fragment } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { compose } from "recompose";
import PropTypes from "prop-types";
import "styles/index.scss";
import * as fromState from "reducers";
import * as appActions from "actions/app";
import Login from "containers/login";
import Main from "components/main";
import Modal from "components/modal";
import { injectIntl, intlShape } from "react-intl";

const App = ({ intl: { formatMessage }, isAuth, shouldShowError, setShowError }) => (
  <Fragment>
    {isAuth ? <Main /> : <Login />}
    <Modal
      isActive={shouldShowError}
      onCloseClick={() => setShowError(false)}
      messageStyle="is-danger"
      title={formatMessage({ id: "Error" })}
      subTitle={formatMessage({ id: "Request failed" })}
    />
  </Fragment>
);

App.propTypes = {
  intl: intlShape.isRequired,
  isAuth: PropTypes.bool.isRequired,
  shouldShowError: PropTypes.bool.isRequired,
  setShowError: PropTypes.func.isRequired,
};

const withConnect = connect(
  state => ({
    isAuth: state.auth.isAuth,
    shouldShowError: fromState.shouldShowError(state),
  }),
  { setShowError: appActions.setShowError },
);

export default compose(
  withRouter,
  withConnect,
  injectIntl,
)(App);
