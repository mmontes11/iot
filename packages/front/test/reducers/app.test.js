import deepFreeze from "deep-freeze";
import appReducer from "reducers/app";
import { TOGGLE_HAMBURGER_MENU, TOGGLE_MAP_DIALOG, SET_SHOW_ERROR } from "constants/actionTypes/app";
import { RESET } from "constants/actionTypes/common";
import { initialState } from "../constants/index";

const appInitialState = initialState.app;
deepFreeze(appInitialState);

describe("reducers/app", () => {
  it("has a default state", () => {
    expect(appReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(appReducer(appInitialState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces TOGGLE_HAMBURGUER_MENU", () => {
    expect(appReducer(appInitialState, { type: TOGGLE_HAMBURGER_MENU })).toMatchSnapshot();
  });
  it("reduces TOGGLE_MAP_DIALOG", () => {
    expect(appReducer(appInitialState, { type: TOGGLE_MAP_DIALOG })).toMatchSnapshot();
  });
  it("reduces SET_SHOW_ERROR", () => {
    expect(appReducer(appInitialState, { type: SET_SHOW_ERROR, showError: true })).toMatchSnapshot();
  });
  it("reduces RESET", () => {
    expect(appReducer(appInitialState, { type: RESET })).toMatchSnapshot();
  });
});
