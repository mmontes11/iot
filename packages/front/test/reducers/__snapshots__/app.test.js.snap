// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/app has a default state 1`] = `
Object {
  "isHamburgerMenuExpanded": false,
  "isMapDialogOpened": false,
  "showError": true,
}
`;

exports[`reducers/app reduces RESET 1`] = `
Object {
  "isHamburgerMenuExpanded": false,
  "isMapDialogOpened": false,
  "showError": true,
}
`;

exports[`reducers/app reduces SET_SHOW_ERROR 1`] = `
Object {
  "isHamburgerMenuExpanded": false,
  "isMapDialogOpened": false,
  "showError": true,
}
`;

exports[`reducers/app reduces TOGGLE_HAMBURGUER_MENU 1`] = `
Object {
  "isHamburgerMenuExpanded": true,
  "isMapDialogOpened": false,
  "showError": true,
}
`;

exports[`reducers/app reduces TOGGLE_MAP_DIALOG 1`] = `
Object {
  "isHamburgerMenuExpanded": false,
  "isMapDialogOpened": true,
  "showError": true,
}
`;

exports[`reducers/app reduces nothing 1`] = `
Object {
  "isHamburgerMenuExpanded": false,
  "isMapDialogOpened": false,
  "showError": true,
}
`;
